### PR TITLE
Added Wayland clipboard support.

### PR DESCRIPTION
**Pull request by Bettehem**
Hello, I added wayland clipboard support using the [wl_clipboard_rs](https://crates.io/crates/wl_clipboard_rs) crate.
Solves [#633](https://github.com/hrkfdn/ncspot/issues/633).

I tested with alacritty in both wayland and xorg and I also tested the primary clipboard and that seems to work as expected too.

### CODE REVIEW

File: src/sharing.rs [hunk 1]
```diff
@@ -1,35 +1,159 @@
...
 #[cfg(not(feature = "share_selection"))]
 pub fn read_share() -> Option<String> {
-    ClipboardProvider::new()
-        .and_then(|mut ctx: ClipboardContext| ctx.get_contents())
-        .ok()
+    if std::option_env!("XDG_SESSION_TYPE").unwrap_or_default() == "wayland"
+        || !std::option_env!("WAYLAND_DISPLAY")
+            .unwrap_or_default()
+            .is_empty()
+        || std::option_env!("GDK_BACKEND").unwrap_or_default() == "wayland"
```
Comment (hrkfdn) [inline src/sharing.rs:25]: Can this be extracted to a separate function for reuse/deduplication?

**Review by hrkfdn — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution :heart:, some minor changes are needed then we can get this in. I have one suggestion in the comments, the other is that this change should not break compilation on macOS and Windows. I haven't tested yet whether this is actually a problem, but I have updated the CI to run checks on both platforms.

File: src/sharing.rs (hunk 1 above)
Comment (Bettehem) [reply src/sharing.rs]: Thanks for your suggestion! I have now added an `is_wayland` function.

**Comment by Bettehem** [comment]
> Thanks a lot for your contribution heart, some minor changes are needed then we can get this in. I have one suggestion in the comments, the other is that this change should not break compilation on macOS and Windows. I haven't tested yet whether this is actually a problem, but I have updated the CI to run checks on both platforms, so you can just rebase on `main` to test this.

I don't have any Windows or macOS machines to test on so I can't test this right now, but I could set up a Windows virtual machine to try this out

**Comment by Bettehem** [comment]
I'm now setting up a windows vm to test building on it. I'll let you know how it goes asap

**Comment by Bettehem** [comment]
I tested on windows now and the build is failing due to wl-clipboard-rs requiring a unix-like os to work. I'll try changing sharing.rs to only use the wl-clipboard-rs crate if a specific feature flag has been set.

**Comment by Bettehem** [comment]
I have now fixed the issue. I also changed the wayland related environment variables to be checked at runtime instead of at compile time so you don't need to recompile when switching from Xorg to Wayland.\
I tested building on Windows with\
`cargo build --no-default-features --features rodio_backend,share_clipboard,cursive/pancurses-backend`\
and
`cargo build --no-default-features --features rodio_backend,share_clipboard,cursive/crossterm-backend`\
Both building and actually using the clipboard works.\
\
I also tested on Linux and if you build with the `wayland_clipboard` feature added, for example:\
`cargo build --features wayland_clipboard`\
\
Using `wayland_clipboard` will compile ncspot with the `wl-clipboard-rs` crate and use it if needed.

**Comment by hrkfdn** [comment]
Merged, thank you!

**Pull request merged**